### DISCUSSION TITLE
Dinit as PID1 on FreeBSD

### DISCUSSION THREAD

**Discussion by hvrsim** [davmac314/dinit#320] [General]
Hello everyone!

Just wondering if using dinit as a PID1 on FreeBSD is possible with the current codebase...

The readme says its not, but the code has *some* support for it? (or atleast I think, didn't really take a deep dive in yet)

If this is not the case, what would be the steps required to achieve it?

**Comment by davmac314** [comment]
There are issues with shutdown. FreeBSD appears to require the root filesystem to be unmounted manually and Dinit's shutdown can't do that yet. (Beyond that, I don't know. It requires testing).

**Reply by hvrsim** [reply]
I see, thanks for the response

**Reply by zoujiaqing** [reply]
谢谢
